### PR TITLE
feat: BotProject File Schema

### LINKED ISSUE #4162: BotProject file (.botProj) Operations
*   [x] Need to support operations to read and load projects from the botproject file in state
*   [x] Add a project to the botproject file
*   [x] Delete a project from the botproject file
*   [x] Update botproject file on "Save as"
*   [x] Adding a remote skill to botproject file
*   [ ] Hosting the approved schema on Bot Framework repo

### PR DESCRIPTION
**Pull request by srinaath**
Version 1.0 Botproject schema. 

This schema supports documenting the Root Bot and the skills data that is required to load the family of bots and start runtimes locally (R11). Version 2.0 would support linking to a Publishing Profile which would encapsulate the deployment configurations required per bot.

This schema would be moved into the SDK repo end of R11

Fixes #4162

### CODE REVIEW

File: Schemas/SampleOfficeVA.botProject.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "https://schemas.botframework.com/schemas/botprojects/v0.1/botproject-schema.json",
+  "name": "OfficeVA",
+  "workspace": "file://../Projects/OfficeVA",
+  "provisioningProfile": "file://../Projects/DeploymentProfiles/officeVA.json",
```
Comment (srinaath) [inline Schemas/SampleOfficeVA.botProject.json:5]: This needs to be publishingProfile

File: Schemas/SampleOfficeVA.botProject.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "skills": [
+    {
+      "workspace": "file://../Projects/RoomSchedulerSkill",
+      "manifest": "room-scheduler-skill-manifest-v2.1.0",
```
Comment (srinaath) [inline Schemas/SampleOfficeVA.botProject.json:9]: Add name of skill to the schema

File: Schemas/botproject-schema.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      "type": "string",
+      "description": "Path to the virtual assistant or Root Bot. Relative to the workspace file or an absolute path."
+    },
+    "provisioningProfile": {
+      "type": "string",
+      "description": "Path to provisioning profile."
+    },
+    "skills": {
+      "type": "array",
+      "minItems": 0,
```
Comment (tdurnford) [inline Schemas/botproject-schema.json:29]: Do we need to have `minItems` if we're setting it to zero?

File: Schemas/botproject-schema.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          "type": "string",
+          "description": "Indication if the skill is remote or local skill.",
+          "$ref": "#/definitions/booleanExpression",
+          "title": "isRemote",
+          "default": false
+        },
+        "endpointName": {
+          "type": "string",
+          "description": "Indicates the name of the skill endpoint to use for writing the skill URL. If skill is remote no need for this property.",
+          "default": "default"
```
Comment (tdurnford) [inline Schemas/botproject-schema.json:62]: If this is optional, do we need a default value?

File: Schemas/botproject-schema.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    },
+    "workspace": {
+      "type": "string",
+      "description": "Path to the virtual assistant or Root Bot. Relative to the workspace file or an absolute path."
+    },
+    "provisioningProfile": {
+      "type": "string",
+      "description": "Path to provisioning profile."
+    },
+    "skills": {
```
Comment (tdurnford) [inline Schemas/botproject-schema.json:27]: I think we should update the 'skills' property to be an object to match how the skills are configured in the settings. It also ensures each skill has a unique name.

```
{
  "skills": {
    "type": "object",
    "additionalProperties": {
      "$ref": "#/definitions/skill"
    }
  }
}
```

File: Composer/packages/server/schemas/botproject.schema [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    "skills": {
+      "description": "List of skills (remote or local) skills that the Root Bot consumes.",
+      "type": "object",
+      "patternProperties": {
+        "^.*$": {
+          "$ref": "#/definitions/skill"
+        }
+      },
+      "additionalProperties": false
```
Comment (srinaath) [inline Composer/packages/server/schemas/botproject.schema:31]:
```json
{
  "patternProperties": {
    "^\\$": { "title": "Skill name", "description": "Skill name identifier" }
  },
  "additionalProperties": { "$ref": "#/definitions/skill" }
}
```

Comment (srinaath) [inline Composer/packages/server/schemas/botproject.schema:31]:
```json
{
  "patternProperties": {
    "^\\$": { "title": "Skill name", "description": "Skill name identifier" }
  },
  "additionalProperties": { "$ref": "#/definitions/skill" }
}
```

Comment (srinaath) [reply Composer/packages/server/schemas/botproject.schema:31]: As per @tdurnford  comment

**Review by cwhitten — APPROVED** [review]

**Pull request merged**